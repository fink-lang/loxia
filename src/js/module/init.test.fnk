{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'
{fink2js} = import '../../testing/generate.fnk'


describe 'module', fn:
  it 'handles comments', fn:
    expect
      fink2js "
        #!/usr/bin/env node

        # leading comment

        # expr comment
        foo = 1234

        # trailing comment
      "
      to_match_snapshot


  it 'handles exports', fn:
    expect
      fink2js "
        delete = fn: true
        Ï€ = fn: true
        foo_bar = 1234
        foo-bar = 1234
      "
      to_match_snapshot