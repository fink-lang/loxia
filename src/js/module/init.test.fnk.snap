// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module handles comments 1`] = `
"#!/usr/bin/env node
const foo_0 = 1234;
export const foo = foo_0;"
`;

exports[`module handles exports 1`] = `
"const delete_0 = () => {
  return true;
};

const π_0 = () => {
  return true;
};

const foo_bar_0 = 1234;
const fooᜭbar_0 = 1234;
export const ˆdelete = delete_0,
      π = π_0,
      foo_bar = foo_bar_0,
      fooᜭbar = fooᜭbar_0;"
`;
